src/parachain/issue.ts

```diff
@@ -206,29 +234,34 @@ export class DefaultIssueAPI extends DefaultTransactionAPI implements IssueAPI {
         }
     }
 
+    async craftRequestTx(
+        vaultId: AccountId,
+        amount: MonetaryAmount<WrappedCurrency, BitcoinUnit>
+    ): Promise<SubmittableExtrinsic<"promise">> {
+        const vault = await this.vaultsAPI.get(vaultId);
+        let griefingCollateral = await this.getGriefingCollateral(amount, vault.collateralCurrency);
```

Comment:
technically, it would help performance if the griefing collateral would be submitted as a parameter to this function as a mapping currencyId -> griefing collateral instead of fetching the griefing collateral rate on-chain for every vault.

Reply:
You can do this in a later PR though!